data_handler/handlers/loan_states/zklend/settings.py

```diff
@@ -20,70 +22,68 @@ class TokenSettings(ZkLendSpecificTokenSettings, TokenSettings):
+    scale_factor = decimal.Decimal("1e27")
+    bigResult = decimal_big_num / scale_factor
+    # Have to cast "0.800000000000000000000000000" to "800000000000000000000000000"
+    formatted_number = "{:.2f}".format(bigResult)
+    return formatted_number
 
-TOKEN_SETTINGS: dict[str, TokenSettings] = {
-    token: TokenSettings(
-        symbol=TOKEN_SETTINGS[token].symbol,
-        decimal_factor=TOKEN_SETTINGS[token].decimal_factor,
-        address=TOKEN_SETTINGS[token].address,
-        collateral_factor=ZKLEND_SPECIFIC_TOKEN_SETTINGS[token].collateral_factor,
-        debt_factor=ZKLEND_SPECIFIC_TOKEN_SETTINGS[token].debt_factor,
-        liquidation_bonus=ZKLEND_SPECIFIC_TOKEN_SETTINGS[token].liquidation_bonus,
-        protocol_token_address=ZKLEND_SPECIFIC_TOKEN_SETTINGS[
-            token
-        ].protocol_token_address,
+def get_value_by_name(data_list, name):
+    """
+    Method to return a specific value in a list based on the name.
+    """
+    return next((item['value'] for item in data_list if item['name'] == name), None)
+
+def create_and_fill_new_zklend_token_setting(reserve_data):
+    """
+    Create and fill new ZkLend token setting.
+    """
+    # Decimal Values
+    collateral_factor = convert_string_to_decimal_1e27(get_value_by_name(reserve_data, 'collateral_factor'))
+    debt_factor = decimal.Decimal("1")
+    liquidation_bonus = convert_string_to_decimal_1e27(get_value_by_name(reserve_data, 'liquidation_bonus'))
+    
+    # STR value
+    protocol_token_address = get_value_by_name(reserve_data, 'z_token_address')
+    
+    return ZkLendSpecificTokenSettings(
+        collateral_factor,
+        debt_factor,
+        liquidation_bonus,
+        protocol_token_address
+    )
+
+async def get_token_reserve_data(token_setting_address):
+    """
+    Make a call to ZKLEND_MARKET_ADDRESSES with the tokenSettingAddress (Address).
+    """
+    reserve_data = await src.blockchain_call.func_call(
+        addr=ProtocolAddresses().ZKLEND_MARKET_ADDRESSES,
+        selector="get_reserve_data",
+        calldata=[token_setting_address],
     )
-    for token in TOKEN_SETTINGS
-}
+    return reserve_data
+
+async def fetch_zklend_specific_token_settings():
+    """
+    Fetch ZkLend specific token settings.
+    """
+    # New dict to store ZkLendSpecificTokenSettings
+    NEW_ZKLEND_SPECIFIC_TOKEN_SETTINGS: dict[str, ZkLendSpecificTokenSettings] = {}
+    
+    # TOKEN_SETTINGS from /derisk-research/data_handler/handlers/settings.py
+    # For each tokenSetting in TOKEN_SETTINGS, get the data from zklend
+    for symbol, token_setting in TOKEN_SETTINGS.items():
+        reserve_data = await get_token_reserve_data(token_setting.address)
+        new_zklend_specific_token_setting = create_and_fill_new_zklend_token_setting(reserve_data)
+        NEW_ZKLEND_SPECIFIC_TOKEN_SETTINGS[symbol] = new_zklend_specific_token_setting
+    
+    return NEW_ZKLEND_SPECIFIC_TOKEN_SETTINGS
```

Comment:
You've deleted `ZKLEND_SPECIFIC_TOKEN_SETTINGS`, replace it's usage with the dynamically loaded token settings.

Reply:
Fix it in this commits, let me know 
bc1e4c29c2cdc13407945fcce5891905bcec461d
0ea14c9337424a2c650c052f628b07e0d22f1f58
I remove the statics used and change it with the constant that is in fetch_zklend_specific_token_setting.py, that have the fetch_zklend_specific_token_setting method call.